src/hist/basehist.py

```diff
@@ -203,7 +210,9 @@ def fill(
         }
 
         if set(data_dict) != set(range(len(args), self.ndim)):
-            raise TypeError("All axes must be accounted for in fill")
+            raise TypeError(
+                "All axes must be accounted for in fill, you may have used a disallowed name in the axes"
```

Comment:
Eventually I'd only do this if you have an axis with a disallowed name, but this is fine for now.

I'd eventually make the `disallowed_names` a global `DISALLOWED_NAMES: Final[set[str]] = {...}`, then we could see if there is any overlap between these, and only suggest the "you may have used a disallowed name in the axes" part if there is a chance of this - most users seeing this error probably are making normal mistakes, not using one of these special names

Reply:
Oh okay, we can make it global after this.